kafkaesk/app.py

```diff
@@ -236,22 +254,21 @@ async def __run(self) -> None:
                         partition=record.partition,
                         group_id=self._subscription.group,
                     ).inc()
-                except UnhandledMessage:
-                    # how should we handle this? Right now, fail hard
-                    logger.warning(f"Could not process msg: {record}", exc_info=True)
-                    # We increase the error counter
+                except Exception as err:
                     CONSUMED_MESSAGES.labels(
                         stream_id=record.topic,
                         partition=record.partition,
-                        error="UnhandledMessage",
+                        error=err.__class__.__name__,
                         group_id=self._subscription.group,
                     ).inc()
-
+                    await retry_policy(record=record, error=err)
```

Comment:
should/can we add a metric here to track the rate and duration of retries?

Reply:
I do need to add metrics for just about everything around retry.  One thing to note here is that the `retry_policy()` call is simply processing the message and potentially forwarding that message along to some other stream.  The message is not actually being passed to the callback again at this point.

That said knowing how long it takes for the message to be processed and scheduled for a retry (or forwarded along to a dead letter stream) would be useful information.

Reply:
I've added metrics to the `retry.py` module.  Would you mind taking a look and letting me know if I've missed anything important?